cpp/include/cudf/hashing/detail/hash_functions.cuh

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2017-2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/utilities/traits.hpp>
+
+#include <limits>
+
+namespace cudf::hashing::detail {
+
+template <typename K>
+struct MurmurHash3_32;
+
+template <typename Key>
+using default_hash = MurmurHash3_32<Key>;
```

Comment:
It’s awkward that this forward declaration forces users to include the MurmurHash3 header just to use the “default.” It injects a requirement of knowing the default so that the header can be included. I wonder if we can reorganize this so that there’s a utility header separate from the header defining the default hash functor. Then we can change the includes so that including the default hash header will include the MurmurHash3 header, which includes the utility header.